src/Config/Validator/Deploy/DeprecatedVariables.php

```diff
@@ -75,44 +65,6 @@ public function validate(): Validator\ResultInterface
             );
         }
 
-        // default value for SCD_EXCLUDE_THEMES is an empty string
-        if (!empty($config[DeployInterface::VAR_SCD_EXCLUDE_THEMES])) {
-            $errors[] = sprintf(
-                'The %s variable is deprecated. Use %s instead.',
-                DeployInterface::VAR_SCD_EXCLUDE_THEMES,
-                DeployInterface::VAR_SCD_MATRIX
-            );
-        }
-
-        if ($this->environment->getEnv(DeployInterface::VAR_STATIC_CONTENT_THREADS)
-            || isset($variables[DeployInterface::VAR_STATIC_CONTENT_THREADS])
-        ) {
-            $errors[] = sprintf(
-                'The %s variable is deprecated. Use %s instead.',
-                DeployInterface::VAR_STATIC_CONTENT_THREADS,
-                DeployInterface::VAR_SCD_THREADS
-            );
-        }
-
-        if ($this->environment->getEnv(DeployInterface::VAR_DO_DEPLOY_STATIC_CONTENT)
-            || isset($variables[DeployInterface::VAR_DO_DEPLOY_STATIC_CONTENT])
-        ) {
-            $errors[] = sprintf(
-                'The %s variable is deprecated. Use %s instead.',
-                DeployInterface::VAR_DO_DEPLOY_STATIC_CONTENT,
-                DeployInterface::VAR_SKIP_SCD
-            );
-        }
-
-        if (isset($config[DeployInterface::VAR_STATIC_CONTENT_SYMLINK])
-            && $config[DeployInterface::VAR_STATIC_CONTENT_SYMLINK] === false
-        ) {
-            $errors[] = sprintf(
-                'The %s variable is deprecated and its behavior will not be supported in the future.',
-                DeployInterface::VAR_STATIC_CONTENT_SYMLINK
-            );
-        }
-
         if ($errors) {
             return $this->resultFactory->error(
```

Comment:
Now you can return `error` in the previous condition, there is no need to keep `$errors` array for the one possible error.

Reply:
Fixed